### PR TITLE
i18n: Update i18n-calypso package to latest version.

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

@see https://github.com/Automattic/i18n-calypso/releases/tag/3.0.0

This release allows us to use translations brought over from Calypso via the SDK:
https://github.com/Automattic/wp-calypso/blob/master/docs/sdk.md

#### Testing instructions:

* Set some other site language in Settings -> General
* Make sure translations still work for Jetpack React
  * To test, go to `/wp-admin/admin.php?page=jetpack#/settings` and verify strings appear translated in the UI
* Make sure translations work for Gutenberg i18n
  * Make sure you have the latest Gutenberg installed.
  * Start a new post `/wp-admin/post-new.php`
  * Run `wp.i18n.__( 'Settings', 'jetpack' )` in your browser console (making sure that this string has a translation for the selected language).
  * Verify a translated value is returned.

I was able to run `yarn build-languages` with no apparent issues after updating. However, it is worth noting that the new strings part of `/_inc/blocks/editor.js` in my local install were not added to `/_inc/jetpack-strings.php`, so we'll still have to figure this out for translations to work.

#### Proposed changelog entry for your changes:
None

### CODE REVIEW

**Comment by kraftbj** [comment]
> However, it is worth noting that the new strings part of /_inc/blocks/editor.js in my local install were not added to /_inc/jetpack-strings.php, so we'll still have to figure this out for translations to work.

For that location, I believe we'd need to update https://github.com/Automattic/jetpack/blob/6.6.1/gulpfile.babel.js#L532 to account for any other locations we want to scrape for translation.

**Review by kraftbj — DISMISSED** [review]
Approving this as is. We can update the gulp file for any new file locations in this PR or a new one.

**Comment by jeherve** [comment]
Shouldn't it get picked up automatically since the blocks live in `/_inc/blocks`?

**Comment by kraftbj** [comment]
No, because it isn't within the `/_inc/client` tree `gulp.src( [ '_inc/client/**/*.js', '_inc/client/**/*.jsx' ] )`

**Comment by jeherve** [comment]
Yup, saw that afterwards, my bad. Sorry!

**Review by tyxla — DISMISSED** [review]
I've done a fresh Jetpack build `yarn distclean && yarn build`, followed by a `yarn build-languages`, and Jetpack React client translations no longer seem to work.

I'm trying with the es_ES locale - see screenshot:



For the protocol, `/wp-content/plugins/jetpack/languages/json/jetpack-es_ES.json` still seems to use the old translation format: 



Are we missing something while building the translations?

**Comment by tyxla** [comment]
FWIW, doing the same on `master` fixes the issue.

**Comment by tyxla** [comment]
@jeherve I've played a little bit here to make this work with the latest i18n-calypso (and respectively, latest Jed) version, and it seems to work correctly now. To do it, I've:

* Updated the gulp `languages:build` task to build translations in `jed 1.x` instead of older `jed`.
* Updated jetpack strings - feel free to delete that one if it doesn't make sense in this PR.
* Updated translation files - feel free to delete that one if it doesn't make sense in this PR. 

I've also rebased the PR.

To test whether this works correctly:
* Set some other site language in Settings -> General
* Make sure translations still work for Jetpack React
  * To test, go to `/wp-admin/admin.php?page=jetpack#/settings` and verify strings appear translated in the UI
* Make sure translations work for Gutenberg i18n
  * Make sure you have the latest Gutenberg installed.
  * Start a new post `/wp-admin/post-new.php`
  * Run `wp.i18n.__( 'Settings', 'jetpack' )` in your browser console (making sure that this string has a translation for the selected language).
  * Verify a translated value is returned.

What we'll want to do in a subsequent PR is as @kraftbj mentioned - to make sure translations in blocks are included when extracting translations in the `languages:extract` gulp task.

**Review by sirreal — COMMENTED** [review]
This tests well for me, nicely done 👍 

With `Español`:

- Editor - `wp.i18n.__( 'Settings', 'jetpack' )` -> `Ajustes` 👍 
- Frontend (view currently enqueues wp-i18n and locales: #10325 ) - `wp.i18n.__( 'Settings', 'jetpack' )` -> `Ajustes` 👍 
- Jetpack dashboard is translated 👍 

I'm unfamiliar with much of how translations work in Jetpack, so consider this approval from a test standpoint, but I'd prefer to leave approval to folks more familiar with the systems in play here.

File: gulpfile.babel.js [hunk 1]
```diff
@@ -454,15 +454,24 @@ gulp.task( 'languages:build', [ 'languages:get' ], function( done ) {
...
 
-					Object.keys( data ).forEach( function( term ) {
+					Object.keys( localeData ).forEach( function( term ) {
 						if ( terms.hasOwnProperty( term ) ) {
-							filtered[ term ] = data[ term ];
+							filtered[ term ] = localeData[ term ];
+
+							// Having a &quot; in the JSON might cause errors with the JSON later
+							if ( typeof filtered[ term ] === 'string' ) {
+								filtered[ term ] = filtered[ term ].replace( '&quot;', '\"' );
```
[the remark this answers is not in the document]
Comment (tyxla) [reply gulpfile.babel.js]: Okay, so here's what happens:

1) There is a `'something'` string in Jetpack.
2) Bulgarian translation changes `'` to `&quot;` for some reason (which is completely valid).
3) We build json files from the translation files.
4) When building, [here](https://github.com/Automattic/jetpack/blob/master/gulpfile.babel.js#L465) the `&quot;` becomes an unescaped double quote for some reason. This seems to be happening when po2json parses the po files.
5) This breaks the JSON and invalidates the translations.

So, to bandaid fix it, we replace `&quot;` with an escaped double quote. 

Let me know if this makes sense.

**Review by brbrr — COMMENTED** [review]
I done some quick round of testing.
`Українська`:
- Jetpack sidebar menu is translated. 
- Jetpack dashboard/settings translated for ~ 1%
- `wp.i18n.__( 'Settings', 'jetpack' ) -> Налаштування` 👌 


`Русский`:
- Jetpack sidebar menu is *not* translated. 
- Jetpack dashboard/settings translated well
- `wp.i18n.__( 'Settings', 'jetpack' ) -> Настройки` 👌 

jetpack jetpack6test mystagingwebsite com at pressable wordpress 2018-10-24 15-56-27

**Comment by jeherve** [comment]
> Updated jetpack strings - feel free to delete that one if it doesn't make sense in this PR.
> Updated translation files - feel free to delete that one if it doesn't make sense in this PR.

To keep things cleaner I merged #10281 which was doing the same thing, and kept only the i18n-calypso related changes in this PR. 

The rebase also allowed me to fix the merge conflicts caused by 0bb294081de898b9709e868249e47aee991f1326

**Review by jeherve — COMMENTED** [review]
Like @brbrr I am seeing some strings missing a translation in Jetpack > Settings (I tested with French, which is fully translated right now). Running `yarn build-languages` does not help.

`wp.i18n.__( 'Settings', 'jetpack' )` does work though.

If I switch back to `master` the strings that were not translated now are. Here is an example of a string that is translated when running `master`, but not when running this branch:
https://github.com/Automattic/jetpack/blob/907874a9d378e59bd6cb4eef5fbefedfd8b68470/_inc/client/discussion/comments.jsx#L99
https://translate.wordpress.org/projects/wp-plugins/jetpack/dev/fr/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=3272819&filters%5Btranslation_id%5D=44908340

The translation is there in `master`: 
https://github.com/Automattic/jetpack/blob/master/languages/json/jetpack-fr_FR.json#L1
And also in this branch:
https://github.com/Automattic/jetpack/blob/update/i18n-calypso/languages/json/jetpack-fr_FR.json#L1

The format is the same for both branches:
`"Comments headline":[null,"Titre du commentaire"]`
`"Comments headline":[null,"Titre du commentaire"]`

I am not sure why the translation won't work. :(

**Comment by jeherve** [comment]
Ok, I made a few mistakes in my testing:

1. You should deactivate the Asset CDN module when testing, to get the local versions of the modified files. 🤦‍♂️ 	
2. If you want to test on Jurassic Ninja instead, you need the updated language files, with the new format, so I'm committing those again so it should be easier to test now.

Sorry about that!

**Review by jeherve — COMMENTED** [review]
It all works for me now with the latest version of this PR and a brand new Jurassic Ninja site! 🎉

**Review by kraftbj — APPROVED** [review]
I confirmed @jeherve's testing. It looks as I'd expect.

**Comment by tyxla** [comment]
Great news, thanks for testing folks! I'm good to 🚢 this, but I'll let one of you do it.

**Pull request merged**